File: sdk-node/examples/openai-completions-claude.ts
import { OpenAI } from "openai";
import { AgentRPC } from "../src";

const openai = new OpenAI({
  apiKey: process.env.ANTHROPIC_API_KEY,
  baseURL: 'https://api.anthropic.com/v1/',
});
const rpc = new AgentRPC({ apiSecret: process.env.AGENTRPC_API_SECRET });


const main = async () => {
  const tools = await rpc.OpenAI.getTools();
    const completion = await openai.chat.completions.create({
        model: "claude-3-sonnet-20240229",
        messages: [
            {
                role: "user",
                content: "What is the weather in Melbourne?",
            },
        ],
        tools,
    });

    const message = completion.choices[0]?.message;

    if (message?.tool_calls) {
        for (const toolCall of message.tool_calls) {
            console.log("Agent is calling Tool", toolCall.function.name);
            const result = await rpc.OpenAI.executeTool(toolCall);
            console.log(result);
        }
    }
};

main();

File: examples/python/openai-completions-claude.py
import os

from agentrpc import AgentRPC
from openai import OpenAI


def main():
    rpc = AgentRPC(
        api_secret=os.environ.get("AGENTRPC_API_SECRET", ""),
    )
    openai = OpenAI(
        api_key=os.environ.get("ANTHROPIC_API_KEY", ""),
        base_url="https://api.anthropic.com/v1/",
    )

    tools = rpc.openai.completions.get_tools()

    completion = openai.chat.completions.create(
        model="claude-3-sonnet-20240229",
        messages=[{"role": "user", "content": "What is the weather in Melbourne?"}],
        tools=tools,
    )

    if completion.choices[0].message.tool_calls:
        for tool_call in completion.choices[0].message.tool_calls:
            print("Agent is calling Tool", tool_call.function.name)
            result = rpc.openai.completions.execute_tool(tool_call)
            print(result)


if __name__ == "__main__":
    main()
